src/main/java/de/unistuttgart/iste/gits/content_service/persistence/dao/UserProgressDataEntity.java

```diff
@@ -17,15 +17,16 @@
 @Builder
 @NoArgsConstructor
 @AllArgsConstructor
-@IdClass(UserProgressDataPrimaryKey.class)
 public class UserProgressDataEntity {
 
     @Id
-    @Column(name = "user_id", nullable = false)
+    @GeneratedValue
+    private UUID id;
```

Comment:
I tried many other things, but this is the only thing that worked: having a new id field for the entities, even though it is completely unecessary